### DISCUSSION TITLE
Buying a computer --- GPUs and CPUs

### DISCUSSION THREAD

**Discussion by oc-rodu** [CliMA/Oceananigans.jl#2765] [General]
A few findings are summarized here which can hopefully serve as a starting point while looking for a new computer. The nature of computer components is that they become obsolete relatively fast, however, it is hoped that some of the ideas should remain helpful even then. 

GPU
There are specialized cards for scientific computing, e.g. Tesla series (perhaps previously called Titan?). This means they excel in computing with 64-bit floating point (FP64). The potential problem is that they are costly. High-end gaming GPUs (GeForce) are much cheaper but they do not do well with double precision (FP64). While using single precision for scientific computing is an active research area, it is still cutting edge and not widespread, also it is not clear how much it can replace double precision computations. Thus, sticking to good FP64 performance seems wise at least for now. 

The RTX series of Nvdia GPU do OK with FP64 at a much lower price than the Tesla series. Another thing to consider is that the Tesla series may not double as a video card (e.g. you may not be able to connect your monitors to it). 

Here are a few resources comparing GPU (search for FP64 if your primary concern is scientific computing)
https://www.microway.com/hpc-tech-tips/nvidia-turing-tesla-t4-hpc-performance-benchmarks/
https://www.gpucheck.com/gpu-benchmark-graphics-card-comparison-chart
https://www.microway.com/knowledge-center-articles/comparison-of-nvidia-geforce-gpus-and-nvidia-tesla-gpus/

One finding was that two  NVIDIA RTX A5000 - 24GB GDDR6 - PCIe 4.0 x16 - Active Cooling (4xDP) with NVLink, was more powerful computationally than one RTX A6000 for a comparable amount of money.

CPU
Just recently AMD Ryzen seemed to be the faster processor, but now it seems Intel has a faster CPU (reaching 6GHz). Things change so fast it will be difficult for any advice to stay accurate, the good news is that there are plenty of resources comparing CPU performance. Other things to consider are how many cores and how well the number of cores interacts with available RAM. It is possible that a faster CPU may perform poorly relative to a slower CPU due to RAM. For more on that topic you can read here (and follow some of the links):
https://www.myroms.org/forum/viewtopic.php?f=17&t=2001&p=7771#p7714
 https://www.myroms.org/forum/viewtopic.php?f=29&t=4665

HTH

**Comment by glwagner** [comment]
I think the Titan V was from a previous generation called "GeForce": https://www.techpowerup.com/gpu-specs/titan-v.c3051. This generation was succeeded by the "Tesla" generation

However, both the Titan V and V100 are based on the "Volta" architecture, according to the same reference: https://www.techpowerup.com/gpu-specs/tesla-v100-pcie-32-gb.c3184

@simone-silvestri @christophernhill do we use the Titan V for graphics? (I think we do.)
